upgrades/toolstorage_test.go

```diff
@@ -56,22 +57,22 @@ func (s *migrateToolsStorageSuite) TestMigrateToolsStorageNoTools(c *gc.C) {
 func (s *migrateToolsStorageSuite) TestMigrateToolsStorage(c *gc.C) {
 	stor := s.Environ.(environs.EnvironStorage).Storage()
 	envtesting.RemoveFakeTools(c, stor, "releases")
-	envtesting.AssertUploadFakeToolsVersions(c, stor, "releases", "released", migrateToolsVersions...)
-	s.testMigrateToolsStorage(c, &mockAgentConfig{})
+	tools := envtesting.AssertUploadFakeToolsVersions(c, stor, "releases", "released", migrateToolsVersions...)
+	s.testMigrateToolsStorage(c, &mockAgentConfig{}, tools)
 }
 
 func (s *migrateToolsStorageSuite) TestMigrateToolsStorageLocalstorage(c *gc.C) {
 	storageDir := c.MkDir()
 	stor, err := filestorage.NewFileStorageWriter(storageDir)
 	c.Assert(err, jc.ErrorIsNil)
-	envtesting.AssertUploadFakeToolsVersions(c, stor, "releases", "released", migrateToolsVersions...)
+	tools := envtesting.AssertUploadFakeToolsVersions(c, stor, "releases", "released", migrateToolsVersions...)
 	for _, providerType := range []string{"local", "manual"} {
 		s.testMigrateToolsStorage(c, &mockAgentConfig{
 			values: map[string]string{
 				agent.ProviderType: providerType,
 				agent.StorageDir:   storageDir,
 			},
-		})
+		}, tools)
```

Comment:
The inlining is fine if there is one arg, but now that you've added a second I'd recommend splitting the config out:

    config := &mockAgentConfig{
        ...
    }
    s.testMigrateToolsStorage(c, config, tools)